src/controllers/aer_controller.hpp

```diff
@@ -887,20 +949,61 @@ Result Controller::execute(std::vector<Circuit> &circuits,
 #endif
     }
 #endif
+
     // then- and else-blocks have intentionally duplication.
     // Nested omp has significant overheads even though a guard condition
     // exists.
     const int NUM_RESULTS = result.results.size();
-    if (parallel_experiments_ > 1) {
-#pragma omp parallel for num_threads(parallel_experiments_)
+    if (parallel_experiments_ > 1 && sim_device_ != Device::GPU && !multi_chunk_req) {
+#pragma omp parallel for if (parallel_experiments_ > 1) num_threads(parallel_experiments_)
       for (int j = 0; j < result.results.size(); ++j) {
-        execute_circuit(circuits[j], noise_model, methods[j],
-                        config, result.results[j]);
+        run_circuit(circuits[j], noise_model,methods[j],
+                    config, circuits[j].shots, circuits[j].seed,
+                    result.results[j],multi_chunk[j]);
       }
-    } else {
-      for (int j = 0; j < result.results.size(); ++j) {
-        execute_circuit(circuits[j], noise_model, methods[j],
-                        config, result.results[j]);
+    }
+    else{
+      bool batch_enable = false;
+      if(batched_shots_optimization_ && batched_shots_optimization_threshold_ >= max_qubits_){
+        if(one_method && sim_device_ == Device::GPU && !multi_chunk_req && circuits.size() > num_gpus_ && 
```

Comment:
We can simplify this test expression by making batch_enable class field.

Reply:
moved to `set_parallelization_experiments()`